packages/qwik/src/core/shared/shared-serialization.ts

```diff
@@ -801,7 +800,13 @@ export const createSerializationContext = (
       } else if (isJSXNode(obj)) {
         discoveredValues.push(obj.type, obj.props, obj.constProps, obj.children);
       } else if (Array.isArray(obj)) {
-        discoveredValues.push(...obj);
+        for (const o of obj) {
```

Comment:
this is wrong, it should just push like before, we need to look at the objects themselves one by one